code/controllers/CMSMain.php

```diff
@@ -1196,16 +1150,23 @@ public function BatchActionList() {
 	
 	public function buildbrokenlinks($request) {
 		// Protect against CSRF on destructive action
-		if(!SecurityToken::inst()->checkRequest($request)) return $this->httpError(400);
+		if(!SecurityToken::inst()->checkRequest($request)) {
+      $this->httpError(400);
+      return;
+    }
```

Comment:
We actually don’t even need the `return` statement here as `RequestHandler::httpError()` throws an `SS_HTTPResponse_Exception`, so it’s redundant. We could keep this on one line and just have:
```php
if(!SecurityToken::inst()->checkRequest($request)) $this->httpError(400);
```